server/test/uploadurl.test.js

```
@@ -0,0 +1,17 @@
+import { describe, it, expect } from 'vitest';
+import { generateUploadUrl } from '../llm.js';
+
+// This test ensures function throws when S3 is not configured
+
+describe('generateUploadUrl', () => {
+  it('throws when S3 env vars missing', async () => {
+    // Clear env vars temporarily
+    const { S3_BUCKET, AWS_REGION } = process.env;
+    delete process.env.S3_BUCKET;
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)